05_stringer/starfallz/main_test.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestMainFunction(t *testing.T) {
+	t.Run("Test main to return string values of IPAddr({127, 0, 0, 1}) with proper formatting", func(t *testing.T) {
+		var buf bytes.Buffer
+		out = &buf
+
+		main()
+
+		expected := strconv.Quote("127.0.0.1")
+		actual := strconv.Quote(buf.String())
+
+		if expected != actual {
+			t.Errorf("Unexpected output, expected: %s, actual: %s", expected, actual)
+		}
+	})
+}
+
+func TestIpAddrDefaultToString(t *testing.T) {
+	testCases := []struct {
+		description string
+		input       IPAddr
+		expected    string
+	}{
+		{"Test IPAddr with all zeros in addresses", IPAddr{0, 0, 0, 0}, "0.0.0.0"},
+		{"Test IPAddr with some values assigned", IPAddr{add1: 5, add4: 4}, "5.0.0.4"},
+		{"Test IPAddr with no value", IPAddr{}, "0.0.0.0"},
+		{"Test IPAddr with max values allowed on uint8", IPAddr{255, 255, 255, 255}, "255.255.255.255"},
```

Comment:
One of the code-owners asked me to have a test case of 'IPAddr{0}" or IPAddr{172, 0},  so perhaps you can try that too?

Reply:
Ok will add them

Reply:
Based on your type declaration for `IPAddr`, I don't think `IPAddr{0}` will compile.

You can partially initialize an array, so `[4]byte{1}` will create an array containing `{1, 0, 0, 0}`, but if you want to initialize a struct without variable identifiers, you need a full list, so `IPAddr{1}` won't compile, but `IPAddr{1, 0, 0, 0}` will

You could write tests with partial struct initialization using the variable identifiers
```
IPAddr{add1: 1, add2:1}
```

Playspace example: https://goplay.space/#uhJY-1G0t3h

Reply:
thanks @anzboi, ive got a test for IPAddr{add1: 1, add2:1} already above. So i guess I will just add another one for {addr1: 0} and {addr1: 172, addr2: 0} thanks!